### PR TITLE
chore(docs): Add link to graphql reference on conceptual guide

### PR DESCRIPTION
**Pull request by gregrickaby**
## Description
Add some more date/time examples. Add a link to Gatsby `date.js`, so users can see all formatting options. Finnally, use May 4th as the example date. #becauseStarWars

## Related Issues
None.

### CODE REVIEW

File: docs/docs/querying-with-graphql.md [hunk 1]
```diff
@@ -147,13 +147,28 @@ GraphQL enables another unique feature of Gatsby — it lets you control data tr
 
 ### Formatting dates
 
-People often store dates like "2018-01-05" but want to display the date in some other form like "January 5th, 2018". One way of doing this is to load a date-formatting JavaScript library into the browser. Or, with Gatsby's GraphQL layer, you can do the formatting at query-time like:
+People often store dates like "2019-05-04" but want to display the date in some other form like "May 4th, 2019". One way of doing this is to load a date-formatting JavaScript library into the browser. Or, with Gatsby's GraphQL layer, you can do the formatting at query-time like:
 
 ```graphql
 {
-  date(formatString: "MMMM Do, YYYY")
+  date(formatString: "MMMM Do, YYYY") // May 4th, 2019
```
Comment (LekoArts) [inline docs/docs/querying-with-graphql.md]: Please remove the comment / undo the change as it's not valid GraphQL syntax

File: docs/docs/querying-with-graphql.md [hunk 2]
```diff
@@ -147,13 +147,28 @@ GraphQL enables another unique feature of Gatsby — it lets you control data tr
...
 }
 ```
+Or maybe add the time:
```
Comment (LekoArts) [inline docs/docs/querying-with-graphql.md]: We can also undo this block and and keep the last line

File: docs/docs/querying-with-graphql.md [hunk 3]
```diff
@@ -147,13 +147,28 @@ GraphQL enables another unique feature of Gatsby — it lets you control data tr
...
+}
+```
+You could even display a machine readible format, which is great for the `<time>` [HTML tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time).
+```graphql
+{
+  date(formatString: "YYYY-MM-DDTHH:mmZ") // 2019-05-04T05:00+00:00
+}
+```
+
+See the full list of formatting options by viewing [the source](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/schema/types/date.js).
```
Comment (LekoArts) [inline docs/docs/querying-with-graphql.md]:
```suggestion
See the full list of formatting options by viewing our [GraphQL Reference page](/docs/graphql-reference/#dates).
```

https://www.gatsbyjs.org/docs/graphql-reference/#dates

Comment (gregrickaby) [reply docs/docs/querying-with-graphql.md]: Oh wow, I didn't know this page existed! 😄 

I even searched "date" and all that came up was "data".

**Comment by gregrickaby** [comment]
@LekoArts  I respectfully disagree. 

The date documentation is hard to find. It's included on a page named [GraphQL Reference](https://www.gatsbyjs.org/docs/graphql-reference/) which, I'm sorry, shouldn't be the source of truth for helpers like dates, excerpts, sorting, etc... That page should be a primer for understanding GraphQL syntax as it relates to Gatsby.

The date formatting information is so difficult to locate, even [Algolia is stumped](https://dl.dropbox.com/s/cpr2lzpv77jv3vh/Screenshot%202019-05-07%2008.33.08.png?dl=0)! 

Adding a couple of examples, or maybe, a more robust date page might be in order. I propose that my examples stay, and I update the comment to proper GraphQL syntax.

**Comment by gregrickaby** [comment]
@LekoArts 

We're after the same goal, quality documentation that's easy to find, scan, and understand. I really appreciate our dialog here.

I'll concede that I'm placing "expanded documentation" in the wrong place, hence your objection to duplicate content. I'm happy to close this PR since the whole reason I opened it, was I didn't know there was another place for Date queries. I do hope the Algolia settings are tweaked to include Date searches. 😄 

In my opinion, the [GraphQL Reference page](https://www.gatsbyjs.org/docs/graphql-reference/) has a LOT of information...and I'd love to discuss improving the scannability of that page. Here is a quick in-browser concept I threw together:
screenshot

What do you think? I could also show this off on tomorrow's Core Maintainer's meeting.

**Comment by LekoArts** [comment]
> I'm happy to close this PR

No! :D I'd like to have a link to the reference page. Could you please change your PR accordingly? Thanks!

> and I'd love to discuss improving the scannability of that page.

Yep, we're on it: https://github.com/gatsbyjs/gatsby/issues/6896

**Comment by gregrickaby** [comment]
@LekoArts I'm getting this error trying to update this PR. Any advice would be 👍 

```bash
remote: error: GH006: Protected branch update failed for refs/heads/gregrickaby-patch-1.        
remote: error: Waiting on code owner review from gatsbyjs/docs.        
To github.com:gatsbyjs/gatsby.git
 ! [remote rejected]     gregrickaby-patch-1 -> gregrickaby-patch-1 (protected branch hook declined)
error: failed to push some refs to 'git@github.com:gatsbyjs/gatsby.git'
```

**Comment by LekoArts** [comment]
@gregrickaby Sorry for the inconveniences, please try again!

**Comment by gregrickaby** [comment]
@LekoArts Same error 🤕

**Comment by LekoArts** [comment]
@gregrickaby Sorry for the troubles! :( Please try again, I'm confident that it now should work :)

**Comment by marcysutton** [comment]
There are more in-depth GraphQL docs improvements planned as it's a key workflow. The first step is to evaluate the current state of the docs for GraphQL in Gatsby as part of #13708, including filing a new issue covering everything found in the evaluation using the grading rubric outlined in the [meta issue](#13708). 

Some things we discussed at the team gathering last week also include renaming that API reference page in the sidebar to reflect what's on it, adding recipes, and improving the docs in general! 

@gregrickaby if you have any interest in helping with the docs evaluation, that would get it moving sooner than later! There are also some relevant TODO items in the [Learning Ecosystem project](https://github.com/gatsbyjs/gatsby/projects/10) that will be filled out as more detailed issues in the coming days.

**Comment by gregrickaby** [comment]
@marcysutton Thank you for the response. I've got all of those links open in tabs, and will review today. 🙌 

@LekoArts Thanks for merging this in! 💪
